framework/db/ActiveQueryTrait.php

```diff
@@ -43,6 +48,18 @@ public function asArray($value = true)
     }
 
     /**
+     * Sets the [[collectionClass]] property.
+     * @param string $className collection class to be instantiated and filled with results.
+     * @return $this the query object itself
+     * @since 2.0.10
+     */
+    public function asCollection($className)
```

Comment:
Default is no collection at all. It doesn't make sense to introduce any default collection since the only thing you can achieve w/ collection that you can't achieve otherwise is type safety.